src/query/observable.js

```diff
@@ -342,6 +350,7 @@ define([
         reset: function (array) {
           if (arguments.length === 0) {
             this.removeAll();
+            this.update();
```

Comment:
I believe this should be removed because the last line in removeAll method is update() and this call will only add overhead to the performance.